UnityGLTF/Assets/UnityGLTF/Scripts/Extensions/GLTFSceneExporter.Data.cs

```diff
@@ -0,0 +1,219 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using UnityEngine;
+using GLTF.Schema;
+using VectorTypes;
+
+namespace UnityGLTF
+{
+	public partial class GLTFSceneExporter
+	{
+
+		private AccessorId ExportData(GLTFAccessorAttributeType type, GLTFComponentType componentType, int componentSize, int count, IEnumerable<object> min, IEnumerable<object> max, int byteLength, Action<BinaryWriter> writeData, string accessorName = null)
+		{
+			// The offset of the data must be aligned to a multiple of the component size.
+			var position = checked((int)this._bufferWriter.BaseStream.Position);
+			var alignedPosition = Align(position, componentSize);
+			for (var i = position; i < alignedPosition; i++)
+			{
+				this._bufferWriter.Write(byte.MinValue);
+			}
+
+			var bufferViewId = this.ExportBufferView(alignedPosition, byteLength);
+			var accessorId = this.ExportAccessor(bufferViewId, componentType, count, type, min, max, accessorName);
+
+			writeData(this._bufferWriter);
+
+			return accessorId;
+		}
+
+		private AccessorId ExportAccessor(BufferViewId bufferViewId, GLTFComponentType componentType, int count, GLTFAccessorAttributeType type, IEnumerable<object> min, IEnumerable<object> max, string name = null)
+		{
+			Accessor accessor = new Accessor
+			{
+				BufferView = bufferViewId,
+				ComponentType = componentType,
+				Count = count,
+				Type = type,
+				Min = (List<double>) min,
+				Max = (List<double>) max,
+				Name = name
+			};
+
+			var id = new AccessorId
+			{
+				Id = _root.Accessors.Count,
+				Root = _root
+			};
+
+			_root.Accessors.Add(accessor);
+
+			return id;
+		}
+
+		private static int Align(int value, int size)
+		{
+			var remainder = value % size;
+			return (remainder == 0 ? value : checked(value + size - remainder));
+		}
+
+		private AccessorId ExportData(IEnumerable<ushort> values)
+		{
+			return this.ExportData(
+				GLTFAccessorAttributeType.SCALAR,
+				GLTFComponentType.UnsignedShort,
+				sizeof(ushort),
+				values.Count(),
+				null,
+				null,
+				sizeof(ushort) * values.Count(),
+				binaryWriter => values.ForEach(value => binaryWriter.Write(value)));
+		}
+
+		private AccessorId ExportData(IEnumerable<float> values, bool minMax = false)
+		{
+			return this.ExportData(
+				GLTFAccessorAttributeType.SCALAR,
+				GLTFComponentType.Float,
+				sizeof(float),
+				values.Count(),
+				minMax ? new object[] { values.Min() } : null,
+				minMax ? new object[] { values.Max() } : null,
+				sizeof(float) * values.Count(),
+				binaryWriter => values.ForEach(value => binaryWriter.Write(value)));
```

Comment:
Shouldn't the binary write take an array of values?

Reply:
in the scene exporter we have right now, we are writing component by component, vector by vector. We can extend it to take in an array though.